### PR TITLE
Refresh parquet readme / contributing guide

### PR DESCRIPTION
**Pull request by alamb**
This PR cleans up the repo readme and contributing guide

I am working on a PR to improve the docs.rs page next https://docs.rs/parquet/8.0.0/parquet/

### CODE REVIEW

File: parquet/README.md [hunk 1]
```diff
@@ -23,38 +23,12 @@
...
 
-## Example
-
-Example usage of reading data:
-
-```rust
-use std::fs::File;
-use std::path::Path;
-use parquet::file::reader::{FileReader, SerializedFileReader};
-
```
Comment (alamb) [inline parquet/README.md:34]: This example is row oriented -- I am working on a PR that has some more realistic examples (and will include them on the docs.rs landing page)

File: parquet/README.md [hunk 2]
```diff
@@ -23,38 +23,12 @@
...
-For an example of reading to Arrow arrays, please see [here](https://docs.rs/parquet/latest/parquet/arrow/index.html)
-
-See [crate documentation](https://docs.rs/parquet/latest/parquet/) for the full API.
+See [crate documentation](https://docs.rs/parquet/latest/parquet/) for examples and the full API.
 
 ## Rust Version Compatbility
 
 This crate is tested with the latest stable version of Rust. We do not currrently test against other, older versions of the Rust compiler.
 
-## Supported Parquet Version
```
Comment (alamb) [inline parquet/README.md:51]: the supported format is already listed in the features section -- moved this content to contributing guide

File: parquet/README.md [hunk 3]
```diff
@@ -63,11 +37,13 @@ version is available. Then simply update version of `parquet-format` crate in Ca
   - [x] Primitive column value readers
   - [x] Row record reader
   - [x] Arrow record reader
+  - [x] Async support (to Arrow)
```
Comment (alamb) [inline parquet/README.md:40]: assumes https://github.com/apache/arrow-rs/pull/1154 is merged

**Review by Jimexist — APPROVED** [review]

**Comment by alamb** [comment]
Parquet doc updates: https://github.com/apache/arrow-rs/pull/1253

**Pull request merged**